### PR TITLE
switch to tests to wast

### PR DESCRIPTION
**Pull request by xtuc**
**What kind of change does this PR introduce?**

The issue is that currently we test against opaque wasm binary and doesn't help when debugging.

I only converted two tests for the demo. I'll continue if you agree.

**Did you add tests for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: lib/WebpackOptionsDefaulter.js [hunk 1]
```diff
@@ -73,6 +73,11 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 			{
 				test: /\.wasm$/i,
 				type: "webassembly/experimental"
+			},
+			{
+				test: /\.wast$/i,
+				loader: "@webassemblyjs/wast-loader",
```
Comment (xtuc) [inline lib/WebpackOptionsDefaulter.js]: I don't know if this is wanted but I think it's a common use case to enable it by default?

File: package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "@webassemblyjs/ast": "1.4.3",
     "@webassemblyjs/wasm-edit": "1.4.3",
     "@webassemblyjs/wasm-parser": "1.4.3",
+    "@webassemblyjs/wast-loader": "^1.5.4",
```
Comment (xtuc) [inline package.json]: To avoid duplicated deps (with other versions), we should align these, but it's already done by some PR i think.

**Comment by xtuc** [comment]
btw I have the `wast-loader` package now. I'm waiting for your review @sokra to continue.

File: lib/WebpackOptionsDefaulter.js [hunk 3]
```diff
@@ -73,6 +73,11 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 			{
 				test: /\.wasm$/i,
 				type: "webassembly/experimental"
+			},
+			{
+				test: /\.wast$/i,
```
Comment (sokra) [inline lib/WebpackOptionsDefaulter.js]: > The recommended file extension for WebAssembly code in textual format is .wat.

https://webassembly.org/docs/text-format/

File: lib/WebpackOptionsDefaulter.js (hunk 1 above)
Comment (sokra) [reply lib/WebpackOptionsDefaulter.js]: Using loaders by default comes with a lot of problems. I would keep it like other loaders.

File: test/cases/wasm/decoding/index.js [hunk 4]
```diff
@@ -1,13 +1,13 @@
 it("should support wasm compiled from c++", function() {
-	return import("./memory3.wasm").then(function(wasm) {
+	return import("./memory3.wast").then(function(wasm) {
```
Comment (sokra) [inline test/cases/wasm/decoding/index.js]: I would keep the `decoding` tests in `wasm` as they try to test special encodings of the wasm binary (like paddings). This is not possible with `wast` (`wat`)

File: package.json (hunk 2 above)
Comment (sokra) [reply package.json]: Should be in devDependencies instead

File: test/cases/wasm/decoding/index.js (hunk 4 above)
Comment (sokra) [reply test/cases/wasm/decoding/index.js]: The other tests can be changed to `.wat` for better readability.

File: lib/WebpackOptionsDefaulter.js (hunk 1 above)
Comment (sokra) [reply lib/WebpackOptionsDefaulter.js]: For the test suite move this into `test/TestCases.template.js`

File: lib/WebpackOptionsDefaulter.js (hunk 3 above)
Comment (xtuc) [reply lib/WebpackOptionsDefaulter.js]: Actually wast (WebAssembly script format) and wat (text format) are two different syntaxes but i agree it's better to switch to `.wat`.

File: test/cases/wasm/decoding/index.js (hunk 4 above)
Comment (xtuc) [reply test/cases/wasm/decoding/index.js]: Good point yes

File: package.json (hunk 2 above)
Comment (xtuc) [reply package.json]: Yes, if we move it from the default options

**Comment by xtuc** [comment]
I kept the LEB128 padded one to ensure we are parsing them, everything else has been changed to wast.

**Comment by xtuc** [comment]
As a side note, wabt's WASM to WAT adds a bit of latency, the test is probably failing because of that. Should we increase the timeout?

**Comment by sokra** [comment]
Thanks

**Pull request merged**